docs/troubleshoot/app-framework/appserver-error-codes.md

```diff
@@ -1800,291 +1800,348 @@ The following error message codes may appear on the app-server log. Use the foll
 
 
 
-### ZWED0063W":"Stack trace follows\n%s
+### ZWED0063W
+
+  Stack trace follows _stackTrace_
 
   **Reason:**
 
-  TODO
+  This exception originates from the web socket and the stack trace message handles the generated exception.
+  The stack trace for an exception helps in understanding the error and what went wrong with the code.
 
   **Action:**
 
-  TODO
+  No action needed unless user is experiencing an interruption in server service, then send stack to developers
 
 
 
-### ZWED0064W":"%s: invalid method %s
+### ZWED0064W
+
+  _plugin.identifier_: Invalid method _method_
 
   **Reason:**
 
-  TODO
+  It will throw the warning if method is invalid (different from these methods: get|post|put|delete|ws)
 
   **Action:**
 
-  TODO
+  Review the warning message and use correct method
+
 
 
+### ZWED0065W
 
-### ZWED0065W":"Library %s is missing libraryVersion attribute for hosting files. Skipping file hosting.
+  Library _plugin.identifier_ is missing libraryVersion attribute for hosting files. Skipping file hosting.
 
   **Reason:**
 
-  TODO
+  Library plugin's plugin definition is missing the library version attribute
 
   **Action:**
 
-  TODO
+  Add the correct library version in the plugin definition
 
 
 
-### ZWED0066W":"%s: getCapabilities() is not a function
+### ZWED0066W
+
+  _pluginID_: _getCapabilities()_ is not a function
 
   **Reason:**
 
-  TODO
+  _getCapabilities()_ is not a formatted variable name in the original message
 
   **Action:**
 
-  TODO
+  No action required. Please send the log and any relevant info to the developers
 
 
 
-### ZWED0068W":"Failed to set proxy authorizations. Error=
+### ZWED0068W
+
+  Failed to set proxy authorizations. Error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  Failed to add proxy authorization with reason _errorMessage_.
 
   **Action:**
 
-  TODO
+  No action required. If your desired authentication plugin isn't successfully authenticating a login,
+  please send the log and any relevant info to the developers.
+
 
 
+### ZWED0069W
 
-### ZWED0069W":"Returning null for cipher array because input had non-string: 
+  Returning null for cipher array because input had non-string:
 
   **Reason:**
 
-  TODO
+  Returns null for cipher array if an array element is not a string type.
 
   **Action:**
 
-  TODO
+  As it is an array of string then all elements should be of type string.
 
 
 
-### ZWED0070W":"Error when reading PFX. Server cannot continue. Error=%s
+### ZWED0070W
+
+  Error when reading PFX. The server cannot continue. Error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  If we get an error while reading _config.https.pfx_ file then the server cannot continue
+  and throws _errorMessage_ in the catch block.
```

Comment:
you can remove "in the catch block" (documentation meant for users to troubleshoot, not programmers)